### PR TITLE
Fix/immutable cache

### PR DESCRIPTION
**Pull request by JAdshead**
Typo in `ImmutableCache#getCacheByKey` causing errors not to be retrieved. This caused an infinite loop where fetchye would continue to attempt to fetch the resource.

### CODE REVIEW

**Comment by JamesSingleton** [comment]
@JAdshead just a nitpick (not a blocker), but some of the tests are `it('should...)` and some are `it('returns...')`.

File: package.json [hunk 1]
```diff
@@ -89,7 +89,7 @@
     "jest": "^26.4.0",
     "lockfile-lint": "^4.3.7",
     "react": "16.x.x",
-    "react-dom": "16.x.x",
+    "react-dom": "^16.14.0",
```
Comment (JamesSingleton) [inline package.json:92]: Why update to a specific version here and not `react`?

**Comment by JAdshead** [comment]
> @JAdshead just a nitpick (not a blocker), but some of the tests are `it('should...)` and some are `it('returns...')`.

I don't agree with `it('should..` waste of time and adds ambiguity.

File: package.json (hunk 1 above)
Comment (JAdshead) [reply package.json:92]: I was attempting to fix the warnings created by react testing library

Comment (JamesSingleton) [reply package.json:92]: @JAdshead do you want to update `react` to match then?

Comment (JAdshead) [reply package.json]: @JamesSingleton i will revert

**Review by JamesSingleton — APPROVED** [review]

File: src/cache/ImmutableCache.js [hunk 2]
```diff
@@ -16,52 +16,13 @@
 
 // Immutable is an optional dependency
 // eslint-disable-next-line import/no-extraneous-dependencies
-import { Map as iMap, Set as iSet } from 'immutable';
```
Comment (infoxicator) [inline src/cache/ImmutableCache.js:19]: bunch of duplicate code? I remember asking Tomcal thought he had removed it? 😅

**Review by infoxicator — APPROVED** [review]

**Review by nellyk — APPROVED** [review]

**Review by Francois-Esquire — APPROVED** [review]

**Pull request merged**